### ISSUE TITLE
Single and double quote are treated as different identifiers

### ISSUE THREAD

**Issue by driskell** [wikimedia/less.php#77]
Single and double quote are treated as different identifiers. This is contrast to javascript based LESS that treats a quoted identifier as a quoted identifier regardless of whether the quotes were single or double. It can cause a little confusion.

As an example:

```
.test-mixin(@value) when (@value = "test") {
  background: red;
}

.test-mixin(@value) when (@value = 'test') {
  background: blue;
}

double {
  .test-mixin("test")
}
single {
  .test-mixin('test')
}
```

Using `wikimedia/less.php` this compiles to the following:

```
double {
  background: red;
}
single {
  background: blue;
}
```

However, using `lessc` Node module you get:

```
double {
  background: red;
  background: blue;
}
single {
  background: red;
  background: blue;
}
```

Using versions:

```
$ lessc --version
lessc 4.1.2 (Less Compiler) [JavaScript]
$ ./vendor/bin/lessc --version
lessc 3.1.0 (less.php)
```

**Comment by driskell** [comment]
Also checked with lessc 2.5.3 which is the listed compatible version for less.php and it's the same as 4.1.2 so seems an incompatibility.

```
body {
  margin: 0;
}
double {
  background: red;
  background: blue;
}
single {
  background: red;
  background: blue;
}
```

**Comment by niket2219** [comment]
Could you assign me this issue .I can fix this.

**Comment by Krinkle** [comment]
@niket2219 Sure, go ahead!

**Comment by niket2219** [comment]
**can you please guide me how to start**

**Comment by Krinkle** [comment]
@niket2219 Sure. I would start by adding a file to your local Git checkout to make testing easier.

For example, create a file `input.less` in your less.php directory like so:

```
.test-mixin(@value) when (@value = "test") {
  background: red;
}

.test-mixin(@value) when (@value = 'test') {
  background: blue;
}

double {
  .test-mixin("test")
}
single {
  .test-mixin('test')
}
```

And then run the following command to confirm what the current behaviour is:

```
/your/path/to/less.php/$ bin/lessc input.less
```

In terms of where to start exploring the bug, the relevant logic resides in the following methods:

* `Less_Parser::parseMixinDefinition -> Less::Parser::parseConditions -> [..] -> Less_Parser::parseEntitiesQuoted`. These methods are responsible for reading the Less input text and understanding what it represents, turning it into Less_Tree objects.
* `Less_Tree_Mixin_Definition` and `Less_Tree_Quoted` are the two objects most relevant to this bug, I suspect.
* Another file that can be of use is `test/assets/less-2.5.3.js` where the upstream JavaScript-based implementation resides, which might be used to perhaps find the difference between two implementations to narrow down the bug.

**Comment by Krinkle** [comment]
Imported to our own issue tracker at: <https://phabricator.wikimedia.org/T357160>.

Fix will be in the next release!

**Comment by Krinkle** [comment]
Released in v4.2.0, https://github.com/wikimedia/less.php/blob/v4.2.0/CHANGES.md
